pkg/logger/logger.go

```diff
@@ -0,0 +1,137 @@
+package logger
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	LogFile       = "$HOME/.config/corectl/log.json"
+	LogMaxSize    = 10
+	LogMaxBackups = 5
+	LogMaxAge     = 30
+	LogCompress   = true
+
+	DefaultTailLines = 20
+
+	TimeFieldKey    = "ts"
+	MessageFieldKey = "status"
+)
+
+var (
+	Log     *zap.Logger
+	LogPath string
+	// AtomicLevel is used to set the terminal log level
+	// Note: we log everything to the log file, but only print
+	//        to the terminal at the level specified here
+	AtomicLevel zap.AtomicLevel
+)
+
+func Init() error {
+	if Log != nil {
+		return nil
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Printf("Failed to get home directory: %v\n", err)
+		os.Exit(1)
```

Comment:
What? Why? What do you need from the user's home dir?|

Reply:
This will not work in containers.

Reply:
pick a better location for logs.

Reply:
The same dir as thie binary or

```sh
/var/log
```

Reply:
It was on the ticket, we use ~/.config/corectl folder throughout